### DISCUSSION TITLE
Is this the only way to use regex?

### DISCUSSION THREAD

**Discussion by cn0047** [go-playground/validator#1018] [Q&A]
I need to validate string like this: `group:__some_name__:eu-main` (`alphanum`, `:`, `-`)
which looks like good candidate for `regex`, and I didn't manage to find anything better than`RegisterValidation`.
My question is: is it possible to come up with something simpler/better than this:
````go
package main

import (
	"fmt"
	"regexp"

	"github.com/go-playground/validator/v10"
)

type Payload struct {
	ResourceName string `json:"resource_name" validate:"required,resource"`
}

var (
	ResourceRegex = regexp.MustCompile(`^([\w:-])+$`)
)

func validateResource(field validator.FieldLevel) bool {
	return ResourceRegex.MatchString(field.Field().String())
}

func main() {
	p := Payload{
		ResourceName: "group:__some_name__:eu-main",
	}

	validate := validator.New()
	err := validate.RegisterValidation("resource", validateResource)
	if err != nil {
		fmt.Printf("failed to register validation, err: %v", err)
	}

	err = validate.Struct(p)
	if err != nil {
		fmt.Printf("validation failed, err: %v", err)
	}

	// ...
}
````

I saw [this issue](https://github.com/go-playground/validator/issues/561) but it's still about `RegisterValidation` which leads to same idea and almost same implementation.

**Comment by zemzale** [comment]
This indeed would be the only way how you could use regex here.
